operator/builtin/input/file/config_test.go

```diff
@@ -204,21 +214,287 @@ func TestConfig(t *testing.T) {
+				return cfg
+			}(),
+		},
+		{
+			"multiline_line_end_special",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				var newMulti *MultilineConfig
+				newMulti = new(MultilineConfig)
+				newMulti.LineEndPattern = "%"
+				cfg.Multiline = newMulti
+				return cfg
+			}(),
+		},
+		{
+			"multiline_random",
+			true,
+			nil,
+		},
+		{
+			"start_at_string",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.StartAt = "beginning"
+				return cfg
+			}(),
+		},
+		{
+			"start_at_empty",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.StartAt = ""
+				return cfg
+			}(),
+		},
+		{
+			"max_concurrent_large",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxConcurrentFiles = 9223372036854775807
+				return cfg
+			}(),
+		},
+		{
+			"max_concurrent_empty",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxConcurrentFiles = 0
+				return cfg
+			}(),
+		},
+		{
+			"max_log_size_mib_lower",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxLogSize = helper.ByteSize(1048576)
+				return cfg
+			}(),
+		},
+		{
+			"max_log_size_mib_upper",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxLogSize = helper.ByteSize(1048576)
+				return cfg
+			}(),
+		},
+		{
+			"max_log_size_mb_upper",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxLogSize = helper.ByteSize(1048576)
+				return cfg
+			}(),
+		},
+		{
+			"max_log_size_mb_lower",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.MaxLogSize = helper.ByteSize(1048576)
+				return cfg
+			}(),
+		},
+		{
+			"max_log_size_invalid_unit",
+			true,
+			nil,
+		},
+		{
+			"encoding_lower",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.Encoding = "utf-16le"
+				return cfg
+			}(),
+		},
+		{
+			"encoding_upper",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.Encoding = "UTF-16lE"
+				return cfg
+			}(),
+		},
 	}
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			cfgFromYaml, err := configFromFileViaYaml(path.Join(".", "testdata", fmt.Sprintf("%s.yaml", tc.name)))
-			require.NoError(t, err)
-			require.Equal(t, tc.expect, cfgFromYaml)
-			cfgFromMapstructure, err := configFromFileViaMapstructure(path.Join(".", "testdata", fmt.Sprintf("%s.yaml", tc.name)))
-			require.NoError(t, err)
-			require.Equal(t, tc.expect, cfgFromMapstructure)
+			cfgFromYaml, err := configFromFileViaYaml(t, path.Join(".", "testdata", fmt.Sprintf("%s.yaml", tc.name)))
+			if tc.expectErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tc.expect, cfgFromYaml)
+			}
+			// TODO cfgFromMapstructure, err := configFromFileViaYaml(t, path.Join(".", "testdata", fmt.Sprintf("%s.yaml", tc.name)))
+			// TODO require.NoError(t, err)
+			// TODO require.Equal(t, tc.expect, cfgFromMapstructure)
```

Comment:
@Mrod1598 This appears to have been reverted. The removed lines are needed to call and validate mapstructure unmarshaling.